lib/teracy-dev-core/providers/docker.rb

```diff
@@ -0,0 +1,22 @@
+require_relative 'provider'
+
+module TeracyDevCore
+  module Providers
+    # see: https://www.vagrantup.com/docs/docker/
+    class Docker < TeracyDevCore::Providers::Provider
+
+      def configure(provider_settings, node_config)
+        options = provider_settings.dup
+
+        ["_id", "type", "enabled"].each do |key|
```

Comment:
move this common stuff to the provider configurator scope: "_id", "type", "enabled", "weight"